core/new-gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -224,18 +263,72 @@ export class ResultPanelComponent {
     this.breakpointAction = false;
   }
 
+  /**
+   * Callback function for table query params changed event
+   *   params containing new page index, new page size, and more
+   *   (this function will be called when user switch page)
+   *
+   * @param params new parameters
+   */
+  public onTableQueryParamsChange(params: NzTableQueryParams) {
+    const { pageSize: newPageSize, pageIndex: newPageIndex } = params;
+    this.currentPageSize = newPageSize;
+    this.currentPageIndex = newPageIndex;
+
+    if (this.isFrontPagination) {
+      return;
+    }
+
+    this.isLoadingResult = true;
+    this.workflowWebsocketService.send('ResultPaginationRequest', { pageSize: newPageSize, pageIndex: newPageIndex });
+    this.workflowWebsocketService.websocketEvent().subscribe(websocketEvent => {
```

Comment:
is `onTableQueryParamsChange` called every time when the pagniation changes? Then you are subscribing to websocket event many times, which is bad. This websocket event subscription only needs to subscribe once.

Reply:
Yes, this method will be called on any pagination change. I will move the the websocket event subscription into the constructor.

Reply:
Fixed in commit https://github.com/Texera/texera/pull/903/commits/04045f42bc57843f1304e1a3533a6e8535786421